File: src/main/java/org/geepawhill/contentment/position/Position.java
package org.geepawhill.contentment.position;

import org.geepawhill.contentment.geometry.*;

import javafx.scene.Node;

public interface Position
{

	Position DEFAULT = new Centered(ViewPort.Companion.getCENTER());

	void position(Node node, PointPair dimensions);
	
}

File: src/main/java/org/geepawhill/contentment/position/Position.kt
package org.geepawhill.contentment.position

import org.geepawhill.contentment.geometry.*

import javafx.scene.Node

interface Position {

    fun position(node: Node, dimensions: PointPair)

    companion object {

        val DEFAULT: Position = Centered(ViewPort.CENTER)
    }

}